hazelcast-client/src/test/java/com/hazelcast/client/config/YamlClientFailoverConfigBuilderTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.util.RootCauseMatcher;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayInputStream;
+import java.net.URL;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class YamlClientFailoverConfigBuilderTest extends AbstractClientFailoverConfigBuilderTest {
+
+    @Rule
+    public ExpectedException expected = ExpectedException.none();
+
+    @Before
+    public void init() throws Exception {
+        URL schemaResource = YamlClientFailoverConfigBuilderTest.class.getClassLoader()
+                                                                      .getResource("hazelcast-client-failover-sample.yaml");
+        fullClientConfig = new YamlClientFailoverConfigBuilder(schemaResource).build();
+    }
+
```

Comment:
Can you add this test case:
two configs, each config partition size is different and we expect an exception?

Reply:
Could you please expand what you mean? We can configure the partition size in the member config. I can't see anything in the client config options with "partition" in its name. Am I missing something?

Reply:
After discussing this with Sancar, I got the point of your comment. Your suggestion would test a feature of the client creation and not the config loading mechanism. This kind of unaccepted asymmetry of the failover client configurations is tested in EE, by `FailoverConfigTest#testClientConfigWith_withAnInvalidChange`.

Reply:
I do not see partition id in this test, is this the correct test or did you mean some other test?

Reply:
@sancar will add for bot xml and yaml.